qemu/tests/virtio_fs_sandbox.py

```diff
@@ -0,0 +1,139 @@
+                                                                    driver_name)
+            # create virtiofs service
+            viofs_svc_name = params["viofs_svc_name"]
+            virtio_fs_utils.create_viofs_service(test, params, session,
+                                                 service=viofs_svc_name)
+
+        fs_target = params["fs_target"]
+        fs_dest = params["fs_dest"]
+        if not is_windows:
+            # mount virtiofs
+            error_context.context("Create a destination directory %s "
+                                  "inside guest." % fs_dest, test.log.info)
+            utils_misc.make_dirs(fs_dest, session)
+            error_context.context("Mount virtiofs target %s to %s inside"
+                                  " guest." % (fs_target, fs_dest),
+                                  test.log.info)
+            if not utils_disk.mount(fs_target, fs_dest, 'virtiofs', session=session):
+                test.fail('Mount virtiofs target failed.')
+
+            # can't change file's owner and group
+            test_file = "%s/test" % fs_dest
+            session.cmd("echo aaa > %s" % test_file)
+            s, o = session.cmd_status_output("chgrp root %s" % test_file)
+            if not s:
+                test.fail("Should not change file's owner/group because"
+                          " it's unprivileged, the output is %s" % o)
+        else:
+            # start virtiofs
+            error_context.context("Start virtiofs service in guest.",
+                                  test.log.info)
+            virtio_fs_utils.start_viofs_service(test, params, session)
+
+            # get fs dest for vm
+            virtio_fs_disk_label = fs_target
+            error_context.context("Get Volume letter of virtio fs target, the disk"
+                                  "lable is %s." % virtio_fs_disk_label,
+                                  test.log.info)
+            vol_con = "VolumeName='%s'" % virtio_fs_disk_label
+            volume_letter = utils_misc.wait_for(
+                lambda: utils_misc.get_win_disk_vol(session, condition=vol_con), cmd_timeout)
+            if volume_letter is None:
+                test.fail("Could not get virtio-fs mounted volume letter.")
+            fs_dest = "%s:" % volume_letter
+
+        # basic io test
+        virtio_fs_utils.basic_io_test(test, params, session)
+    finally:
+        if is_windows:
+            virtio_fs_utils.delete_viofs_serivce(test, params, session)
+            # for windows guest, disable/uninstall driver to get memory leak based on
+            # driver verifier is enabled
+            win_driver_utils.memory_leak_check(vm, test, params)
+        else:
+            utils_disk.umount(fs_target, fs_dest, 'virtiofs', session=session)
+            utils_misc.safe_rmdir(fs_dest, session=session)
+        if vm and vm.is_alive():
+            vm.destroy()
+            process.system(params["del_user_cmd"], ignore_status=True, shell=True)
```

Comment:
```suggestion
        process.system(params["del_user_cmd"], ignore_status=True, shell=True)
```
Actually, the framework would help us to clean up the VM instance:
https://github.com/avocado-framework/avocado-vt/blob/5b4a4eb93a318d5f095dfcfdea421f2b2d0f8847/virttest/env_process.py#L628-L632
```
    if params.get("kill_vm") == "yes":
        kill_vm_timeout = float(params.get("kill_vm_timeout", 0))
        if kill_vm_timeout:
            utils_misc.wait_for(vm.is_dead, kill_vm_timeout, 0, 1)
        vm.destroy(gracefully=params.get("kill_vm_gracefully") == "yes")
```

Reply:
Thanks. Add vm destroy here just because we need to make sure qemu-kvm process is killed before delete the common user.

Reply:
> Thanks. Add vm destroy here just because we need to make sure qemu-kvm process is killed before delete the common user.

Thanks, I got your concern, and I have a question we will not delete the common user if not vm.is_alive(), right?
Since base on your code: I understand that we just only delete the common user in the condition of the vm is alive, we don't clean up the common user in other conditions.

Reply:
To be precise, virtiofsd is booted up with a common user, only if it quits the user can be deleted, and virtiofs process will quit when the vm quit.

Reply:
> To be precise, virtiofsd is booted up with a common user, only if it quits the user can be deleted, and virtiofs process will quit when the vm quit.

Get it, thanks for explaining.